test/operator-framework-e2e/operator_framework_test.go

```diff
@@ -0,0 +1,970 @@
+	opmPath            = "../../bin/opm"
+	deployedNameSpace  = "rukpak-system"
+)
+
+func TestOperatorFramework(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Operator Framework E2E Suite")
+}
+
+var _ = BeforeSuite(func() {
+
+	var err error
+	cfg = ctrl.GetConfigOrDie()
+
+	scheme := runtime.NewScheme()
+
+	err = catalogd.AddToScheme(scheme)
+	Expect(err).ToNot(HaveOccurred())
+
+	err = operatorv1alpha1.AddToScheme(scheme)
+	Expect(err).To(Not(HaveOccurred()))
+
+	err = rukpakv1alpha1.AddToScheme(scheme)
+	Expect(err).To(Not(HaveOccurred()))
+
+	c, err = client.New(cfg, client.Options{Scheme: scheme})
+	Expect(err).To(Not(HaveOccurred()))
+
+	ctx = context.Background()
+
+})
+
+var _ = Describe("Operator Framework E2E for plain bundles", func() {
+	var (
+		bundleInfo      *BundleInfo
+		catalogDInfo    *CatalogDInfo
+		operatorAction  *OperatorActionInfo
+		operatorCatalog *catalogd.Catalog
+		operator        *operatorv1alpha1.Operator
+		err             error
+	)
+	BeforeEach(func() {
+		bundleInfo = &BundleInfo{
+			baseFolderPath: "../../testdata/bundles/plain-v0",
+			bundles: []BundleContent{
+				{
+					bInputDir:     "plain.v0.1.0",
+					bundleVersion: "0.1.0",
+				},
+				{
+					bInputDir:     "plain.v0.1.1",
+					bundleVersion: "0.1.1",
+				},
+			},
+		}
+		catalogDInfo = &CatalogDInfo{
+			baseFolderPath:     "../../testdata/catalogs",
+			fbcFileName:        "catalog.yaml",
+			operatorName:       "plain",
+			desiredChannelName: "beta",
+		}
+		operatorAction = &OperatorActionInfo{
+			installVersion: "0.1.0",
+			upgradeVersion: "0.1.1",
+		}
+		for i, b := range bundleInfo.bundles {
+			bundleInfo.bundles[i].imageRef = remoteRegistryRepo + catalogDInfo.operatorName + "-bundle:v" + b.bundleVersion
+		}
+		catalogDInfo.catalogDir = catalogDInfo.operatorName + "-catalog"
+		catalogDInfo.imageRef = remoteRegistryRepo + catalogDInfo.catalogDir + ":test"
+	})
+	When("Build and load plain+v0 bundle images into the test environment", func() {
+		It("Build the plain bundle images and load them", func() {
+			for _, b := range bundleInfo.bundles {
+				dockerContext := bundleInfo.baseFolderPath + "/" + b.bInputDir
+				dockerfilePath := dockerContext + "/Dockerfile"
+				err = buildPushLoadContainer(b.imageRef, dockerfilePath, dockerContext, kindServer, GinkgoWriter)
+				Expect(err).To(Not(HaveOccurred()))
+			}
+		})
+	})
+	When("Create the FBC", func() {
+		It("Create a FBC", func() {
+			By("Creating FBC for plain bundle using custom routine")
+			var imageRefs []string
+			var bundleVersions []string
+			for _, b := range bundleInfo.bundles {
+				imageRefs = append(imageRefs, b.imageRef)
+				bundleVersions = append(bundleVersions, b.bundleVersion)
+			}
+			fbc := CreateFBC(catalogDInfo.operatorName, catalogDInfo.desiredChannelName, imageRefs, bundleVersions)
+			err = WriteFBC(*fbc, catalogDInfo.baseFolderPath+"/"+catalogDInfo.catalogDir, catalogDInfo.fbcFileName)
+			Expect(err).To(Not(HaveOccurred()))
```

Comment:
Just a minor style nitpick but perhaps makes these consistently `Expect(err).ToNot(HaveOccurred())` like you have it in some other places.